### PR TITLE
Update HorizontalScroll component

### PR DESCRIPTION
**Pull request by PavelHolec**
image

image

Pagination:
https://user-images.githubusercontent.com/35844477/161776206-0f5f3329-ee2b-42ad-bd5d-266270f1511d.mov

### CODE REVIEW

File: Sources/Orbit/Components/HorizontalScroll.swift [hunk 1]
```diff
@@ -16,6 +16,15 @@ public enum HorizontalScrollItemHeight {
     case custom(CGFloat)
 }
 
+struct HorizontalScrollWidthKey: PreferenceKey {
```
Comment (sjavora) [inline Sources/Orbit/Components/HorizontalScroll.swift:19]: This looks like an implementation details, but it's sandwiched between `public` types.

Comment (PavelHolec) [reply Sources/Orbit/Components/HorizontalScroll.swift:19]: Where to put it though - public goes first - and they cannot be nested in the generic type.

Comment (PavelHolec) [reply Sources/Orbit/Components/HorizontalScroll.swift:19]: Where to put them though. They cannot go inside the component, so I at least ordered it by public first.

Comment (sjavora) [reply Sources/Orbit/Components/HorizontalScroll.swift:19]: Above previews?

Comment (PavelHolec) [reply Sources/Orbit/Components/HorizontalScroll.swift]: Updated - removed the specific key and unified with others.

**Review by sjavora — APPROVED** [review]

**Pull request merged**